src/v/kafka/server/connection_context.cc

```diff
@@ -42,6 +42,16 @@ ss::future<> connection_context::process_one_request() {
           if (!sz) {
               return ss::make_ready_future<>();
           }
+          if (sz > _max_request_size()) {
```

Comment:
Also possibly the commit message can be changed. Looks like `shutdown_input` doesn't drop the connection, because `close()` will not be called on the stream. Close is called in the case exceptions are thrown from this context and handled by `net::server`

Reply:
see the comment above